WordPress/src/main/java/org/wordpress/android/ui/reader/discover/ReaderDiscoverViewModel.kt

```diff
@@ -188,9 +166,41 @@ class ReaderDiscoverViewModel @Inject constructor(
         }
     }
 
+    private suspend fun convertCardsToUiStates(posts: ReaderDiscoverCards): List<ReaderCardUiState> {
+        return posts.cards.map {
+            when (it) {
+                is ReaderPostCard -> postUiStateBuilder.mapPostToUiState(
+                        post = it.post,
+                        isDiscover = true,
+                        photonWidth = photonWidth,
+                        photonHeight = photonHeight,
+                        isBookmarkList = false,
+                        onButtonClicked = this@ReaderDiscoverViewModel::onButtonClicked,
+                        onItemClicked = this@ReaderDiscoverViewModel::onPostItemClicked,
+                        onItemRendered = this@ReaderDiscoverViewModel::onItemRendered,
+                        onDiscoverSectionClicked = this@ReaderDiscoverViewModel::onDiscoverClicked,
+                        onMoreButtonClicked = this@ReaderDiscoverViewModel::onMoreButtonClicked,
+                        onMoreDismissed = this@ReaderDiscoverViewModel::onMoreMenuDismissed,
+                        onVideoOverlayClicked = this@ReaderDiscoverViewModel::onVideoOverlayClicked,
+                        onPostHeaderViewClicked = this@ReaderDiscoverViewModel::onPostHeaderClicked,
+                        onTagItemClicked = this@ReaderDiscoverViewModel::onTagItemClicked,
+                        postListType = TAG_FOLLOWED
+                )
+                is InterestsYouMayLikeCard -> {
+                    postUiStateBuilder.mapTagListToReaderInterestUiState(
+                            it.interests,
+                            this@ReaderDiscoverViewModel::onReaderTagClicked
+                    )
+                }
+            }
+        }
+    }
+
     private fun onReaderTagClicked(tag: String) {
-        val readerTag = readerUtilsWrapper.getTagFromTagName(tag, FOLLOWED)
-        _navigationEvents.postValue(Event(ShowPostsByTag(readerTag)))
+        launch(bgDispatcher) {
+            val readerTag = readerUtilsWrapper.getTagFromTagName(tag, FOLLOWED)
```

Comment:
Since `getTagFromTagName` isn't a suspendable function I decided to start this coroutine on a bg thread.

Reply:
Yep, those are database calls. We started using ioDispatcher for other reader database calls (in usecases), do we want to do the same here? wdyt?

Reply:
Sounds good, these are all database calls. I wonder if we want to use Dispatchers.IO? We started using IO for database work in `Get` usecases. wdyt?

Reply:
Fixed in https://github.com/wordpress-mobile/WordPress-Android/pull/12806/commits/c4648919c2747c18707d58804c4a974bffbf5496